crates/primitives/src/bytecode.rs

```diff
@@ -66,10 +73,23 @@ impl Bytecode {
         matches!(self, Self::Eof(_))
     }
 
+    /// Checks if the bytecode starts with the EOF prefix followed by the correct version.
+    #[inline]
+    pub fn is_eof_format(bytes: &Bytes) -> bool {
+        bytes.len() >= 2 && bytes[0] == EOF_PREFIX && bytes[1] == EOF_VERSION
```

Comment:
This is not correct. `EOF_MAGIC` is 0xEF00 while `EOF_VERSION` (At least version one) is 0x01. Would be good to move those in revm-primitives crate

prefix is `0xEF0001`